src/test/java/teammates/sqlui/webapi/PublishFeedbackSessionActionTest.java

```diff
@@ -0,0 +1,170 @@
+import teammates.ui.webapi.EntityNotFoundException;
+import teammates.ui.webapi.JsonResult;
+import teammates.ui.webapi.PublishFeedbackSessionAction;
+
+/**
+ * SUT: {@link PublishFeedbackSessionAction}.
+ */
+public class PublishFeedbackSessionActionTest extends BaseActionTest<PublishFeedbackSessionAction> {
+
+    private Course course1;
+    private FeedbackSession feedbackSession1;
+    private FeedbackSession feedbackSession2;
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.SESSION_PUBLISH;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return POST;
+    }
+
+    @BeforeMethod
+    void setUp() throws InvalidParametersException, EntityDoesNotExistException {
+        course1 = generateCourse1();
+        feedbackSession1 = generateSession1InCourse(course1);
+        feedbackSession2 = generateSession2InCourse(course1);
+        Instructor instructor1 = generateInstructor1InCourse(course1);
+
+        when(mockLogic.getFeedbackSession(feedbackSession1.getName(), course1.getId())).thenReturn(feedbackSession1);
+        when(mockLogic.publishFeedbackSession(
+                feedbackSession1.getName(), course1.getId())).thenReturn(feedbackSession2);
+        when(mockLogic.getInstructorByGoogleId(
+                course1.getId(), instructor1.getAccount().getGoogleId())).thenReturn(instructor1);
+
+        loginAsInstructor(instructor1.getAccount().getGoogleId());
+    }
+
+    @Test
+    protected void testExecute() {
+        ______TS("Typical case");
+
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, course1.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, feedbackSession1.getName(),
+        };
+
+        PublishFeedbackSessionAction publishFeedbackSessionAction = getAction(params);
+
+        JsonResult result = getJsonResult(publishFeedbackSessionAction);
+        FeedbackSessionData feedbackSessionData = (FeedbackSessionData) result.getOutput();
+
+        when(mockLogic.getFeedbackSession(feedbackSession1.getName(), course1.getId())).thenReturn(feedbackSession2);
+
+        assertEquals(feedbackSessionData.getFeedbackSessionName(), feedbackSession1.getName());
+        assertEquals(FeedbackSessionPublishStatus.PUBLISHED, feedbackSessionData.getPublishStatus());
+        assertTrue(mockLogic.getFeedbackSession(feedbackSession1.getName(), course1.getId()).isPublished());
+
+        ______TS("Typical case: Session is already published");
+        // Attempt to publish the same session again.
+
+        result = getJsonResult(getAction(params));
+        feedbackSessionData = (FeedbackSessionData) result.getOutput();
+
+        assertEquals(feedbackSessionData.getFeedbackSessionName(), feedbackSession1.getName());
+        assertEquals(FeedbackSessionPublishStatus.PUBLISHED, feedbackSessionData.getPublishStatus());
+        assertTrue(mockLogic.getFeedbackSession(feedbackSession1.getName(), course1.getId()).isPublished());
```

Comment:
For future PRs can split these into separate test cases. One of the reasons the __TS tag was used for integration tests in the past was that we only need to set up the data for the tests once. Now with unit tests, it is more beneficial to split this into smaller unit tests.